sleap/gui/suggestions.py

```diff
@@ -195,29 +203,36 @@ def prediction_score(
 
     @classmethod
     def _prediction_score_video(
-        cls, video: Video, labels: "Labels", score_limit: float, instance_limit: int
+        cls,
+        video: Video,
+        labels: "Labels",
+        score_limit: float,
+        instance_limit_upper: int,
+        instance_limit_lower: int,
     ):
         lfs = labels.find(video)
         frames = len(lfs)
-        idxs = np.ndarray((frames), dtype="int")
-        scores = np.full((frames, instance_limit), 100.0, dtype="float")
+        # initiate an array filled with -1 to store frame index (starting from 0).
+        idxs = np.full((frames), -1, dtype="int")
 
-        # Build matrix with scores for instances in frames
         for i, lf in enumerate(lfs):
-            # Scores from instances in frame
-            frame_scores = [inst.score for inst in lf if hasattr(inst, "score")]
-            # Just get the lowest scores
-            if len(frame_scores) > instance_limit:
-                frame_scores = sorted(frame_scores)[:instance_limit]
-            # Add to matrix
-            scores[i, : len(frame_scores)] = frame_scores
-            idxs[i] = lf.frame_idx
-
-        # Find instances below score of <score_limit>
-        low_instances = np.nansum(scores < score_limit, axis=1)
-
-        # Find all the frames with at least <instance_limit> low scoring instances
-        result = idxs[low_instances >= instance_limit].tolist()
+            # Scores from visible instances in frame
+            pred_fs = lf.instances_to_show
+            frame_scores = np.array(
+                [inst.score for inst in pred_fs if hasattr(inst, "score")]
+            )
+            # Gets the number of instances with scores lower than <score_limit>
+            n_qualified_instance = np.nansum(frame_scores <= score_limit)
+
+            if (
+                n_qualified_instance >= instance_limit_lower
```

Comment:
`instance_limit_lower` has a default of 0, which will evaluate to `True` for empty frames I think. Like:

```py
frame_scores = np.array([])  # == no predicted instances in frame
n_qualified_instance = np.nansum(frame_scores <= score_limit)  # == 0
```

Do we maybe want the lower limit to be 1 or switch to using `>` instead of `>=`?

Reply:
That's a good point. I guess one of the uses of this feature is to quickly locate the frames where prediction failed, and that would include empty frames (e.g. after a whole video prediction). Therefore, I prefer to keep the option open to users